interactions/api/gateway.py

```diff
@@ -297,17 +297,26 @@ def handle_dispatch(self, event: str, data: dict) -> None:
                         _name = context.data.custom_id
                     elif data["type"] == InteractionType.APPLICATION_COMMAND_AUTOCOMPLETE:
                         _name = "autocomplete_"
-                        if hasattr(context.data, "options"):
+                        if "options" in context.data:
                             if context.data.options:
                                 for option in context.data.options:
-                                    if option["type"] in (
-                                        OptionType.SUB_COMMAND,
-                                        OptionType.SUB_COMMAND_GROUP,
-                                    ):
-                                        if option.get("options"):
+                                    if option["type"] == OptionType.SUB_COMMAND_GROUP:
```

Comment:
I think we could build a submethod to make this check easier inside of the `handle_dispatch`:

```py
def check_sub_auto(option):
    if "options" in option:
        if option["type"] == OptionType.SUB_COMMAND:
            if sub_option.get("focused") for sub_option in option["options"]:
                _args.append(option["name"])
        elif option["type"] == OptionType.SUB_COMMAND_GROUP:
            if "options" in group_option for group_option in option["options"]:
                if sub_option.get("focused") for sub_option in group_option["options"]:
                    _name += sub_option["name"]
                    _args.append(sub_option["value"])
    elif option.get("focused"):
        _name += option{"name"]
        _args.append(option["value"])
```

This check will also respect the issue you have with changed line 307, where we suddenly ignore the actual group option. Black should also format all of this to respect the character line limit, so it theoretically will look better once pre-committed.